src/interfaces.ts

```diff
@@ -59,7 +59,33 @@ export enum AuthenticationTypes {
 
 export type AuthenticationType = `${AuthenticationTypes}`;
 
-export type AuthenticationOptions = { apiKeyHeaderName?: string };
+interface CommonSecurityDefinition {
+  id: string;
+}
+
+export interface BasicSecurityDefinition extends CommonSecurityDefinition {
+  type: AuthenticationTypes.BASIC;
+}
+
+export interface ApiKeySecurityDefinition extends CommonSecurityDefinition {
+  type: AuthenticationTypes.API_KEY;
+  header: string;
+}
+
+export type SecurityDefinition =
+  | BasicSecurityDefinition
+  | ApiKeySecurityDefinition;
```

Comment:
This union doesn't behave quite as nicely as I would like where it requires explicit type definitions more than I would have liked (see `tests/auth.test.ts` where I have to explicitly define `BasicSecurityDefinition` and `SecurityDefinition[]` to get it to behave... maybe there is a better approach.